frameworks/react-webpack5/src/types.ts

```diff
@@ -1,49 +1,46 @@
-import type ForkTsCheckerWebpackPlugin from 'fork-ts-checker-webpack-plugin';
-import type { Configuration } from 'webpack';
+import type {
+  ReactOptions,
+  StorybookConfig as StorybookConfigBase,
+  TypescriptOptions as TypescriptOptionsReact,
+} from '@storybook/preset-react-webpack';
+import type {
+  StorybookConfigWebpack,
+  BuilderOptions,
+  TypescriptOptions as TypescriptOptionsBuilder,
+} from '@storybook/builder-webpack5';
 
-import type { StorybookConfig as BaseStorybookConfig } from '@storybook/preset-react-webpack';
+type FrameworkName = '@storybook/react-webpack5';
+type BuilderName = '@storybook/builder-webpack5';
 
-export type { BuilderResult } from '@storybook/preset-react-webpack';
-
-/**
- * Options for TypeScript usage within Storybook.
- */
-export type TypescriptOptions = BaseStorybookConfig['typescript'] & {
-  /**
-   * Configures `fork-ts-checker-webpack-plugin`
-   */
-  checkOptions: ForkTsCheckerWebpackPlugin['options'];
+export type FrameworkOptions = ReactOptions & {
+  builder?: BuilderOptions;
 };
 
-export interface StorybookTypescriptConfig {
-  typescript?: Partial<TypescriptOptions>;
-}
-
-export interface ReactOptions {
-  fastRefresh?: boolean;
-  strictMode?: boolean;
-  /**
-   * Use React's legacy root API to mount components
-   * @description
-   * React has introduced a new root API with React 18.x to enable a whole set of new features (e.g. concurrent features)
-   * If this flag is true, the legacy Root API is used to mount components to make it easier to migrate step by step to React 18.
-   * @default false
-   */
-  legacyRootApi?: boolean;
-}
-
-export interface StorybookReactConfig {
+type StorybookConfigFramework = {
```

Comment:
Should we be consistent and call this `FrameworkStorybookConfig`?

Reply:
We should be consistent, I think naming it like `StorybookConfigX` is preferred. That's what I've been trying to do everywhere.

Reply:
I guess I was contrasting with `FrameworkName` and `FrameworkOptions` above, but I guess they are slightly different.